### PR TITLE
Add SQL Definition for `ALTER SESSION` of Oracle Database

### PR DESCRIPTION
**Pull request by ThanoshanMV**
#10111

Hi @tristaZero, @wgy8283335 I've added SQL definition for [ALTER SESSION](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ALTER-SESSION.html#GUID-27186B28-7EFC-4998-B1ED-2B905CC0211B). Please check it. I'll change based on your feedback. 

Changes proposed in this pull request:
- Added SQL definition for `ALTER SESSION`
- Added test cases for `ALTER SESSION`

### CODE REVIEW

**Comment by wgy8283335** [comment]
I'd like to review this pr.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 1]
```diff
@@ -998,3 +998,91 @@ supplementalLogGrpClause
...
+
+alterSessionOption
+    : adviseClause
+    | closeDatabaseLinkClause
+    | commitInProcedureClause
+    | securiyClause
+    | parallelExecutionClause
+    | resumableClause
+    | shardDdlClause
+    | SYNC WITH PRIMARY
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:1014]: Why not put `SYNC WITH PRIMARY` in a rule?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 2]
```diff
@@ -998,3 +998,91 @@ supplementalLogGrpClause
...
+    | containerClause
+    | rowArchivalVisibilityClause
+    | defaultCollationClause
+    ;
+
+parameterClause
+    : (parameterName EQ_ parameterValue)+
+    ;
+
+parameterValue
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:1070]: Why `parameterValue` is identified like this? and maybe put it in BaseRule.g4 is better.

**Review by wgy8283335 — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 (hunk 2 above)
Comment (wgy8283335) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:1070]: Why `parameterValue` is identified as `literals | identifier`?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 (hunk 1 above)
Comment (ThanoshanMV) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:1014]: Yes, we can put it in a rule. I'll put it in a rule like: 
```
syncWithPrimaryClause: SYNC WITH PRIMARY
```

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 (hunk 2 above)
Comment (ThanoshanMV) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:1070]: Okay @wgy8283335, I'll put `parameterValue` in BaseRule.g4.

Based on [ALTER SESSION documentation](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ALTER-SESSION.html#GUID-27186B28-7EFC-4998-B1ED-2B905CC0211B), we can set two types of parameters: [initialization](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ALTER-SESSION.html#GUID-8DBA8659-413E-49B4-98D3-D9608C9C8026) and [session parameters](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ALTER-SESSION.html#GUID-DC7B8CDD-4F89-40CC-875F-F70F673711D4). 

[Basic initialization parameters](https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/basic-initialization-parameters.html#GUID-D75F1A77-47E2-4F35-B145-44B3A10ED85C) such as [CLUSTER_DATABASE](https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/CLUSTER_DATABASE.html#GUID-DA4B6C11-FFEA-46BE-82D7-13C1DBD36277)'s parameter value is boolean,  [DB_BLOCK_SIZE](https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/DB_BLOCK_SIZE.html#GUID-6DD6D804-D03D-4C57-A764-826EE0F3836B)'s parameter value is an integer. 

[Session parameters](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ALTER-SESSION.html#GUID-DC7B8CDD-4F89-40CC-875F-F70F673711D4) such as `INSTANCE`'s parameter value is an integer, `CURRENT_SCHEMA` is an identifier, `TIME_ZONE` is having a string literal as the parameter value. 

To include all of this, I specified the `parameterValue` as `literals | identifier`. I just checked for which values they can contain. Please correct me if I'm wrong.

Comment (wgy8283335) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:1070]: Thank you for your explanation. I've got it.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 (hunk 1 above)
Comment (wgy8283335) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:1014]: Good. After you finished, tell me, please.

Comment (ThanoshanMV) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4]: I've finished it @wgy8283335

**Review by wgy8283335 — APPROVED** [review]

**Comment by wgy8283335** [comment]
@tristaZero shall we merge the pr?

**Comment by tristaZero** [comment]
> @tristaZero shall we merge the pr?

Merged. Thanks everyone.

**Pull request merged**